crates/types/src/net/codec.rs

```diff
@@ -42,8 +42,8 @@ impl<T> WireEncode for Box<T>
 where
     T: WireEncode,
 {
-    fn encode_to_bytes(self, protocol_version: ProtocolVersion) -> Bytes {
-        (*self).encode_to_bytes(protocol_version)
+    fn encode_to_bytes(&self, protocol_version: ProtocolVersion) -> Bytes {
+        (**self).encode_to_bytes(protocol_version)
```

Comment:
I think this reads better

```suggestion
self.as_ref().encode_to_bytes(protocol_version)
```